android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java

```diff
@@ -104,7 +104,13 @@ void updateAndSend() {
             if (network != null) {
                 // This may return null per API docs, and is deprecated, but for older APIs (< VERSION_CODES.P)
                 // we need it to test for suspended internet
-                networkInfo = getConnectivityManager().getNetworkInfo(network);
+                try {
+                    networkInfo = getConnectivityManager().getNetworkInfo(network);
+                } catch (SecurityException e) {
+                    // Android 11 may throw a 'package does not belong' security exception here.
+                    // We need to catch this to prevent app crash.
```

Comment:
```suggestion
                // Android 11 may throw a 'package does not belong' security exception here.
                // Google fixed Android 14, 13 and 12 with the issue where Chaland Jean patched those versions.
                // Android 11 is too old, so that's why we have to catch this exception here to be safe.
               //  https://android.googlesource.com/platform/frameworks/base/+/249be21013e389837f5b2beb7d36890b25ecfaaf%5E%21/
```